llama_stack/providers/inline/agents/meta_reference/persistence.py

```diff
@@ -191,8 +191,13 @@ async def list_sessions(self) -> list[Session]:
         sessions = []
         for value in values:
             try:
-                session_info = Session(**json.loads(value))
-                sessions.append(session_info)
+                session_data = json.loads(value)
+                if "session_name" not in session_data:
```

Comment:
this conditional will always trigger right? 

can you add a unit / integration test for this. it looks like this was just plain broken? 

cc @ehhuang @franciscojavierarceo